jplag.endToEndTesting/src/test/java/jplag/endToEndTesting/javaTestCases.java

```diff
@@ -0,0 +1,242 @@
+				System.out.println("Comparison of the stored values and the current equality values");
+				assertEquals(resultComparison.getSimilarity(), jPlagComparison.similarity());
+			}
+			// after close the created directories are deleted
+
+		} catch (Exception e) {
+			fail(e.getMessage());
+		}
+	}
+
+	/**
+	 * Inserting comments or empty lines (normalization level)
+	 */
+	@Test
+	void normalizationLevelTest_three() {
+		String[] testClassNames = new String[] { "SortAlgo1.java", "SortAlgo2.java" };
+
+		try {
+			var testSuiteHelper = new JPlagTestSuiteHelper(testClassNames);
+
+			List<String> tempList = new ArrayList<String>();
+			tempList.add(testSuiteHelper.getFolderPath());
+
+			JPlagOptions options = new JPlagOptions(tempList, new ArrayList<String>(), LanguageOption.JAVA);
+
+			JPlag jplag = new JPlag(options);
+			JPlagResult result = jplag.run();
+			// after close the created directories are deleted
+			testSuiteHelper.close();
+
+			ResultComparison resultComparison = new ResultComparison(new Object() {
+			}.getClass().getEnclosingMethod().getName());
+			// for the fine-granular testing strategy, the result object contains only one
+			// comparison.
+			for (JPlagComparison jPlagComparison : result.getAllComparisons()) {
+				System.out.println("Comparison of the stored values and the current equality values");
+				assertEquals(resultComparison.getSimilarity(), jPlagComparison.similarity());
+			}
+
+		} catch (Exception e) {
+			fail(e.getMessage());
+		}
+	}
+
+	/**
+	 * Insertion of unnecessary or changed code lines (token generation)
+	 */
+	@Test
+	void tokenGenerationLevelTest_one() {
+		String[] testClassNames = new String[] { "SortAlgo.java", "SortAlgo3.java" };
+
+		try {
+			var testSuiteHelper = new JPlagTestSuiteHelper(testClassNames);
+
+			List<String> tempList = new ArrayList<String>();
+			tempList.add(testSuiteHelper.getFolderPath());
+
+			JPlagOptions options = new JPlagOptions(tempList, new ArrayList<String>(), LanguageOption.JAVA);
+
+			JPlag jplag = new JPlag(options);
+			JPlagResult result = jplag.run();
+			// after close the created directories are deleted
+			testSuiteHelper.close();
+
+			ResultComparison resultComparison = new ResultComparison(new Object() {
+			}.getClass().getEnclosingMethod().getName());
+			// for the fine-granular testing strategy, the result object contains only one
+			// comparison.
+			for (JPlagComparison jPlagComparison : result.getAllComparisons()) {
+				System.out.println("Comparison of the stored values and the current equality values");
+				assertEquals(resultComparison.getSimilarity(), jPlagComparison.similarity());
+			}
+
+		} catch (Exception e) {
+			fail(e.getMessage());
+		}
+	}
+
+	/**
+	 * Changing the program flow (token generation) (statments and functions must be
+	 * independent from each other)
+	 */
+	@Test
+	void tokenGenerationLevelTest_two() {
+		String[] testClassNames = new String[] { "SortAlgo.java", "SortAlgo4.java" };
+
+		try {
```

Comment:
why try catch in a test case?

Reply:
Was for debug reasons but will be removed in the right PR. I will take it out next time right there you are right

Reply:
done